posts/map_blocks_example.md

```diff
@@ -0,0 +1,191 @@
+    SALT = SALT.reset_coords(drop=True)
+    TEMP = TEMP.reset_coords(drop=True)
+
+    # compute density
+    rho = pop_tools.eos(SALT.chunk({'z_t': 10}),
+                        TEMP.chunk({'z_t': 10}),
+                        depth=SALT.z_t * 0.).compute()
+
+    # add these coordinates which creep in somewhere, I think when xarray does the unstack
+    # without these, I get an error: not expecting {'nlat', 'nlon'}
+    # chunking here is arbitrary and maybe suitable for the global POP_gx1v7 grid
+    if 'nlat' in rho.dims and 'nlon' in rho.dims:
+        rho = rho.assign_coords({
+            'nlat': xr.DataArray(np.arange(len(SALT.nlat)), dims=('nlat')),
+            'nlon': xr.DataArray(np.arange(len(SALT.nlon)), dims=('nlon')),
+        })
+
+    rho = rho.chunk(rho_chunks).persist()
+
+    # Setup a template
+    template = rho.isel(z_t=0).drop('z_t')
+    template.attrs['long_name'] = 'MLD'
+    template.attrs['units'] = SALT.z_t.attrs['units']
+    template.name = 'MLD'
+
+    # compute and return MLD - this is where the parallelization comes in
+    return xr.map_blocks(
+        interp_mld_dsigma, rho,
+        kwargs=dict(dsigma=dsigma),
+        template=template,
+    )
+```
+
+### Applying Mixed Layer Depth Calculation
+
+This function, `interp_mld_dsigma` is designed to be called within `xr.map_blocks`. It assumes that `rho_in` has a depth dimension, `z_t` and some number of unspecified other dimensions, `non_vertical_dims`.
+
+The non_vertical_dims are "stacked" using `xarray.stack` and the code loops over these dimensions, performing linear interpolation in `z_t` at each location.
+
+The data are returned unstacked with the dimensions ordered as they arrived in `rho_in`.
+
+We define the function here, then immediately call it to test it; arbitrarily, we're calling it on TEMP rather than density.
+
+```python
+def interp_mld_dsigma(rho_in, dsigma=0.03):
+    """compute MLD at point using interpolation"""
+
+    non_vertical_dims = [d for d in rho_in.dims if d not in ['z_t']]
+    rho_stack = rho_in.stack(non_vertical_dims=non_vertical_dims)
+    mld_stack = xr.full_like(rho_stack.isel(z_t=slice(0, 1)), fill_value=np.nan)
+    N = len(rho_stack.non_vertical_dims)
+    z_t = rho_in.z_t
+
+    for i in range(N):
+        # if all NaN, skip this column
+        if np.isnan(rho_stack[:, i]).all():
+            continue
+
+        # if the whole column has density less than the threshold, set MLD to deepest point
+        if (rho_stack[:, i] < rho_stack[0, i] + dsigma).all():
+            k = np.where(~np.isnan(rho_stack[:, i]))[0]
+            mld_stack[:, i] = z_t[k[-1]]
+
+        # linearly interpolate
+        else:
+            f = interpolate.interp1d(
+                rho_stack[:, i], z_t,
+                assume_sorted=False,
+            )
+            mld_stack[:, i] = f(rho_stack[0, i] + dsigma)
+
+    return mld_stack.unstack().isel(z_t=0, drop=True).transpose(*non_vertical_dims)
```

Comment:
```suggestion
    return mld_stack.unstack().transpose(*non_vertical_dims)
```

Reply:
This requires further changes in the loop below (all instances of `mld_stack[:, i]` need to be changed `mld_stack[i]`